### PR TITLE
feat: Bazel expanding support beyond WORKSPACE

### LINKED ISSUE #3413: Support Bazel dependencies declared in files other than WORKSPACE
**What would you like Renovate to be able to do?**
One common pattern seen in a lot of Bazel repositories is to declare a list of external dependencies of a project in a macro (example https://github.com/bazelbuild/rules_docker/blob/968e0b7c8b3bc7e009531231ac926325cd2745bc/repositories/repositories.bzl#L55).

With such a pattern, in order to import the project I linked to, you can do:
```
http_archive(
    name = "io_bazel_rules_docker",
    ...
)

load(
    "@io_bazel_rules_docker//repositories:repositories.bzl",
    container_repositories = "repositories",
)
container_repositories()
```

I would like renovate to update http_archive, git_repository, etc declarations it finds in *any* *.bzl file, not just the WORKSPACE file so that these dependencies are updated as well.

**Describe the solution you'd like**
Include all *.bzl files when searching for Bazel targets that need to be updated.

**Describe alternatives you've considered**
No viable alternatives as far as I know. We are just updating these dependencies manually for now.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Dmitry9**
[Issue](https://github.com/renovatebot/renovate/issues/3413)
Added one more file-match regex to capture files with extension `bzl`.
Tested against [test-repo](https://github.com/Dmitry9test/renovate-test/blob/master/rules_docker/repositories/repositories.bzl)
see  lib/workers/repository/extract/file-match.js:45
`matchedFiles` holds an array which second element is the full file path from tested repo. This file is expected to be captured.

Closes #3413

### CODE REVIEW

**Comment by Dmitry9** [comment]
@rarkins can I proceed to the second stage:
`verify that existing extract and update functions are flexible enough with white space or if they need improving`?

**Comment by rarkins** [comment]
@Dmitry9 	yes please proceed. We don't want to match these additional files unless our extract/update can work with them

**Comment by Dmitry9** [comment]
@rarkins please check out `lib/manager/bazel/extract.js`. Do I on the correct way?

**Comment by rarkins** [comment]
@Dmitry9 	I don't understand all of the code or why you're doing it, so it's hard for me to say if it's the correct track. Importantly, does it now parse both old and new file types?

**Comment by Dmitry9** [comment]
All tests are passed.
Test repo updating [PR](https://github.com/Dmitry9test/renovate-test/pull/9)
This [file](https://github.com/Dmitry9test/renovate-test/blob/master/rules_docker/repositories/repositories.bzl) is splitted to strings, each starts with  `http_archive=(` 
......................
ends with `)` .
From debug.log:
```JSON
DEBUG: Found 8 definitions (repository=Dmitry9test/renovate-test)
       "definitions": [
         "http_archive(\n            name = \"bazel_source\",\n            sha256 = \"3f5441ab5bf403bc44e31bf018ed994d02a1bd2dbf350eface5bf343bc7dff4c\",\n            strip_prefix = \"bazel-1cf75b9dab3e0bb2ffa3014d6e657aa486b1c7bf\",\n            urls = [(\"https://github.com/bazelbuild/bazel/archive/1cf75b9dab3e0bb2ffa3014d6e657aa486b1c7bf.tar.gz\")],\n        )",
         "http_archive(\n            name = \"containerregistry\",\n            sha256 = \"98a7d40b7b45dc76f031c9e17728dddb963f8ec28a1ee4d18693e57155d198f8\",\n            strip_prefix = \"containerregistry-\" + CONTAINERREGISTRY_RELEASE[1:],\n            urls = [(\"https://github.com/google/containerregistry/archive/\" +\n                     CONTAINERREGISTRY_RELEASE + \".tar.gz\")],\n        )",
         "http_archive(\n            name = \"httplib2\",\n            build_file_content = \"\"\"\npy_library(\n   name = \"httplib2\",\n   srcs = glob([\"**/*.py\"]),\n   data = [\"cacerts.txt\"],\n   visibility = [\"//visibility:public\"]\n)\"\"\",\n            sha256 = \"d9f568c183d1230f271e9c60bd99f3f2b67637c3478c9068fea29f7cca3d911f\",\n            strip_prefix = \"httplib2-0.11.3/python2/httplib2/\",\n            type = \"tar.gz\",\n            urls = [\"https://codeload.github.com/httplib2/httplib2/tar.gz/v0.11.3\"],\n        )",
         "http_archive(\n            name = \"six\",\n            build_file_content = \"\"\"\n# Rename six.py to __init__.py\ngenrule(\n    name = \"rename\",\n    srcs = [\"six.py\"],\n    outs = [\"__init__.py\"],\n    cmd = \"cat $< >$@\",\n)\npy_library(\n   name = \"six\",\n   srcs = [\":__init__.py\"],\n   visibility = [\"//visibility:public\"],\n)\"\"\",\n            sha256 = \"e24052411fc4fbd1f672635537c3fc2330d9481b18c0317695b46259512c91d5\",\n            strip_prefix = \"six-1.9.0/\",\n            type = \"tar.gz\",\n            urls = [\"https://pypi.python.org/packages/source/s/six/six-1.9.0.tar.gz\"],\n        )",
         "http_archive(\n            name = \"oauth2client\",\n            build_file_content = \"\"\"\npy_library(\n   name = \"oauth2client\",\n   srcs = glob([\"**/*.py\"]),\n   visibility = [\"//visibility:public\"],\n   deps = [\n     \"@httplib2//:httplib2\",\n     \"@six//:six\",\n   ]\n)\"\"\",\n            sha256 = \"7230f52f7f1d4566a3f9c3aeb5ffe2ed80302843ce5605853bee1f08098ede46\",\n            strip_prefix = \"oauth2client-4.0.0/oauth2client/\",\n            type = \"tar.gz\",\n            urls = [\"https://codeload.github.com/google/oauth2client/tar.gz/v4.0.0\"],\n        )",
         "http_archive(\n            name = \"concurrent\",\n            build_file_content = \"\"\"\npy_library(\n   name = \"concurrent\",\n   srcs = glob([\"**/*.py\"]),\n   visibility = [\"//visibility:public\"]\n)\"\"\",\n            sha256 = \"a7086ddf3c36203da7816f7e903ce43d042831f41a9705bc6b4206c574fcb765\",\n            strip_prefix = \"pythonfutures-3.0.5/concurrent/\",\n            type = \"tar.gz\",\n            urls = [\"https://codeload.github.com/agronholm/pythonfutures/tar.gz/3.0.5\"],\n        )",
         "http_archive(\n            name = \"subpar\",\n            sha256 = \"7ab6ab37ede82255e00c0456846a1428b20e8813f77d83bcf54ddd59ba34377a\",\n            # Commit from 2019-03-07.\n            strip_prefix = \"subpar-0356bef3fbbabec5f0e196ecfacdeb6db62d48c0\",\n            urls = [\"https://github.com/google/subpar/archive/0356bef3fbbabec5f0e196ecfacdeb6db62d48c0.tar.gz\"],\n        )",
         "http_archive(\n            name = \"bazel_skylib\",\n            sha256 = \"eb5c57e4c12e68c0c20bc774bfbc60a568e800d025557bc4ea022c6479acc867\",\n            strip_prefix = \"bazel-skylib-0.6.0\",\n            urls = [\"https://github.com/bazelbuild/bazel-skylib/archive/0.6.0.tar.gz\"],\n        )"
       ]
```
`bazel_skylib` was detected and updated.
Parsed .bsl file contain multyline python comments. I do not exclude those comments when split file content to 8 definitions.
File also contain toplevel variable
[CONTAINERREGISTRY_RELEASE](https://github.com/Dmitry9test/renovate-test/blob/master/rules_docker/repositories/repositories.bzl#L27) which holds value that needs to be dynamicaly insirted into difinitions. This is not implemented yet.
@rarkins please take a look.

**Comment by Dmitry9** [comment]
Backward compatability notes:
Look into lib/manager/bazel/extract.js:69 
I excluded from regexp `http_archive` string
was: 
`/(go_repository|git_repository|http_archive)\(([\s\S]*?)\n\)\n?/g`
now:
`/(go_repository|git_repository)\(([\s\S]*?)\n\)\n?/g`
This means that all other defenitions (not http_archive-type) will be parsed as before.
 http_archive-type defenitions captured [here](https://github.com/renovatebot/renovate/pull/3459/files#diff-b938aae7ff4c76430daddfb81feb0990R72).
`httpDefs` [variable](https://github.com/renovatebot/renovate/pull/3459/files#diff-b938aae7ff4c76430daddfb81feb0990R86) holds array of already trancated strings each constitute a single http_archive-type definition.

**Comment by rarkins** [comment]
Why is `http_archive`special? Won’t we need to handle all the different types inside bzl files?

**Comment by rarkins** [comment]
@Dmitry9 	is there any reason why you can't use the same logic for all types we currently detect?  e.g. including go_registry and git_registry?

**Comment by Dmitry9** [comment]
@rarkins the only reason why I do not use the same logic for all types is to make my change as isolated as it can be (diff is small and easy to reason about) before I get your approval. If you want it all in the same PR please direct me to the definition of  go_registry and git_registry. For example:
Drawing (1)
By giving file name and green border I have all I need to implement parser http_archive-type def. In this case for simplicity I chose def without inner comments or inclosing parentheses.
The example `bzl` file also contains a top-level variable
[CONTAINERREGISTRY_RELEASE](https://github.com/Dmitry9test/renovate-test/blob/master/rules_docker/repositories/repositories.bzl#L27) which holds a value that needs to be dynamically inserted into definitions. This is not implemented yet.

**Comment by rarkins** [comment]
Inserting variables should not be in the scope for this PR. But supporting all types we currently support can be included here.

**Comment by Dmitry9** [comment]
@rarkins please direct me to the definition of go_registry and git_registry

**Comment by rarkins** [comment]
It will use the same fields as we already support

**Comment by Dmitry9** [comment]
[Extraction from bzl file](https://github.com/renovatebot/renovate/pull/3459/files#diff-25b5f8ee723137ac4797499b29088511R3)

**Comment by Dmitry9** [comment]
@rarkins please take a look

**Comment by rarkins** [comment]
@Dmitry9 do any changes need to be made for the update to work? Could you add some tests with the new bazel fixture to make sure?

**Comment by Dmitry9** [comment]
@rarkins no changes need to be made for the update to work
`tests with the new bazel fixture`: I added test/manager/bazel/_fixtures/repositories.bzl

**Comment by rarkins** [comment]
@Dmitry9 can you add an update test to https://github.com/Dmitry9/renovate/blob/feat/%233413-Support-Bazel-dependencies-declared-in-files-other-than-WORKSPACE/test/manager/bazel/update.spec.js that updates the new fixture? So far I think it's only tested with `extract.spec.js`

File: test/manager/bazel/_fixtures/repositories.bzl
Comment (rarkins) [inline test/manager/bazel/_fixtures/repositories.bzl]: Please add some tests to `update.spec.js` that reference this fixture

**Review by rarkins — CHANGES_REQUESTED** [review]

File: test/manager/bazel/_fixtures/repositories.bzl
Comment (Dmitry9) [reply test/manager/bazel/_fixtures/repositories.bzl]: added test to the `test/manager/bazel/update.spec.js`

**Comment by Dmitry9** [comment]
@rarkins I have updated `test/manager/bazel/update.spec.js`.

**Pull request merged**
